### PR TITLE
AB#101753: Webhooks backend

### PR DESCRIPTION
**Pull request by AndyRae**
## Overview

Adds the initial backend for implementing Webhooks, allowing a new webhook to be created, with corresponding trigger parameters and filters.

### CODE REVIEW

**Review by MoizIqbal97 — DISMISSED** [review]

**Review by dcl10 — APPROVED** [review]

File: app/Decsys/Controllers/WebhooksController.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            return BadRequest(ModelState);
+        }
+
+        try
+        {
+            var webhookId  = _webhooks.Create(model);
+            return Ok(webhookId);
+        }
+        catch (Exception)
```
Comment (blargly) [inline app/Decsys/Controllers/WebhooksController.cs:40]: Why is this being caught in the controller if we're just throwing a 500 anyway? Or are we intentionally removing the exception details?

**Review by blargly — APPROVED** [review]

**Review by MoizIqbal97 — DISMISSED** [review]

File: app/Decsys/Models/Webhooks/WebhookModel.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    /// <summary>
+    /// The actual page the user is navigated to.
+    /// </summary>
+    public int ResolvedPage;
+    
+    /// <summary>
+    /// Whether the navigation was successful.
+    /// </summary>
+    public bool ResolvedSuccess;
```
Comment (dcl10) [inline app/Decsys/Models/Webhooks/WebhookModel.cs:54]: Do these not need the `{get; set;}` to work?

Comment (beforan) [reply app/Decsys/Models/Webhooks/WebhookModel.cs:54]: "Work" not necesarily, but yeah we do want them to be properties, not fields. Good catch.

Comment (AndyRae) [reply app/Decsys/Models/Webhooks/WebhookModel.cs]: Added, thanks Dan!

**Review by dcl10 — APPROVED** [review]

**Pull request merged**